### DISCUSSION TITLE
Please HELP!! grid_remove() is not working!

### DISCUSSION THREAD

**Discussion by moyurun0321** [TomSchimansky/CustomTkinter#1433] [Q&A]
I tried grid removal, and this came up.

I have all of my frames in a class and I am trying to remove an element from a button in a different class. I am new to coding and I searched google, but I could not find an answer that works.

File "C:\Users\moyur\AppData\Local\Programs\Python\Python311\Lib\tkinter_init_.py", line 2565, in grid_remove
self.tk.call('grid', 'remove', self._w)
^^^^^^^
AttributeError: type object 'Data' has no attribute 'tk'

**Comment by Akascape** [comment]
> I am new to coding and I searched google

If you are a beginner then learn to do debugging and basics about class object.

Just giving the error info can't describe the issue everytime. Atleast try to provide some code of the block/line causing the issue.

**Reply by moyurun0321** [reply]
I will try to do debugging and I will try to learn more about the class.  For now, should I just cover it up with placeholder texts?

**Reply by moyurun0321** [reply]
def f1():
            Suggestions.Suggestions = Suggestions(self)
            Suggestions.configure(self, height=500, width=500)
            Suggestions.Suggestions.grid(columnspan=4, row=5, column=0, padx=10, pady=10)

        def f2():
            Data.Data = Data(self)
            Data.configure(self, height=500, width=500)
            Data.Data.grid(columnspan=4, row=5, column=0, padx=10, pady=10)

        def f3():
            Learn.Learn = Learn(self)
            Learn.configure(self, height=500, width=500)
            Learn.Learn.grid(columnspan=4, row=5, column=0, padx=10, pady=10)

this is the part where it is giving the error

**Reply by Akascape** [reply]
Still can't figure out, 😕
Your class objects are looking so weird.
Learn about python class properly.

**Reply by Akascape** [reply]
You may find a solution here:
https://stackoverflow.com/questions/27946030/tkinter-attributeerror-object-has-no-attribute-tk

**Reply by moyurun0321** [reply]
thank you

**Reply by avalon60** [reply]
If you use an IDE (e.g. Pycharm - Community edition is free) debugging should be relatively easy and there are videos on YouTube covering this.
Your code looks strange. You appear to be defining regular subroutines but there are no  class declarations, even though you appear to be trying to reference a class object (self). 

By the way, you should only use camelcase when declaring a class. Use lower case for instances of class objects as well as regular functions and variables.

**Reply by moyurun0321** [reply]
sorry, it's because I did not put the part where I declared it because the entire thing is 500 lines long.
